File: 3222/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(losingPlayer(2, 7))
	fmt.Println(losingPlayer(4, 11))
}

func losingPlayer(x int, y int) string {
	f := false
	for x >= 1 && y >= 4 {
		f = !f
		x--
		y -= 4
	}
	if f {
		return "Alice"
	}
	return "Bob"
}

File: 3222/main.rs
struct Solution {}

impl Solution {
    pub fn losing_player(x: i32, y: i32) -> String {
        let num = x.min(y/4);
        if num % 2 == 0 {
            return "Bob".to_string();
        }
        "Alice".to_string()
    }
}

fn main() {
    println!("{}", Solution::losing_player(4, 11));
    println!("{}", Solution::losing_player(2, 7));
}